Assets/Scripts/UI/Menus/PauseMenu.cs

```diff
@@ -0,0 +1,105 @@
+﻿using Managers;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+using UnityEngine.UI;
+
+namespace UI.Menus
+{
+    public class PauseMenu : MonoBehaviour
+    {
+        [Tooltip("The content of the pause menu")] [SerializeField]
+        private GameObject pauseContent;
+
+        [Tooltip("The content of the options menu")] [SerializeField]
+        private GameObject optionsContent;
+
+        [Tooltip("The background dim (black, semi-transparent image)")] [SerializeField]
+        private GameObject backgroundDim;
+
+        [Tooltip("Whether the game should be frozen/stopped onPause")] [SerializeField]
+        private bool freezeGameOnPause;
+
+        [Header("Buttons")] [Tooltip("The button for continuing the game")] [SerializeField]
+        private Button continueButton;
+
+        [Tooltip("The button for opening the options")] [SerializeField]
+        private Button optionsButton;
+
+        [Tooltip("The button for leaving the options")] [SerializeField]
+        private Button optionsBackButton;
+
+        [Tooltip("The button for going back to the main menu")] [SerializeField]
+        private Button mainMenuButton;
+
+        [Tooltip("The button for exiting the game")] [SerializeField]
+        private Button quitButton;
+
+
+        public delegate void PauseDelegate(bool isPaused);
+
+        /// <summary>
+        /// Will be invoked when the game gets paused
+        /// </summary>
+        public static PauseDelegate OnPause;
```

Comment:
I think there's an `event` missing here ^^ Not obligatory, but better code quality, I'd say :)